homeassistant/components/neevo/__init__.py

```diff
@@ -0,0 +1,130 @@
+"""The Nee-Vo Tank Monitoring integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from aiohttp.client_exceptions import ClientError
+from pyneevo import NeeVoApiInterface
+from pyneevo.errors import (
+    GenericHTTPError,
+    InvalidCredentialsError,
+    InvalidResponseFormat,
+    PyNeeVoError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import COORDINATOR, DOMAIN, TANKS
+
+_LOGGER = logging.getLogger(__name__)
+
+_PUSH_UPDATE = "neevo.push_update"
+_INTERVAL = 60
+
+_PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Nee-Vo Tank Monitoring from a config entry."""
+
+    email = config_entry.data[CONF_EMAIL]
+    password = config_entry.data[CONF_PASSWORD]
+
+    try:
+        api = await NeeVoApiInterface.login(email, password=password)
+    except InvalidCredentialsError:
+        _LOGGER.error("Invalid credentials provided")
+        return False
+    except PyNeeVoError as err:
+        _LOGGER.error("Config entry failed: %s", err)
+        raise ConfigEntryNotReady from err
+
+    try:
+        tanks = await api.get_tanks_info()
+    except (ClientError, GenericHTTPError, InvalidResponseFormat) as err:
+        raise ConfigEntryNotReady from err
+
+    async def fetch_update():
+        """Fetch the latest changes from the API."""
+        try:
+            return await api.get_tanks_info()
+        except PyNeeVoError as err:
+            raise UpdateFailed(err) from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="Neevo",
+        update_method=fetch_update,
+        update_interval=timedelta(minutes=_INTERVAL),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = {
+        COORDINATOR: coordinator,
+        TANKS: tanks,
```

Comment:
probably no need to add it to the config_entry, as you can retrieve it from the coordinator